src/qt/walletmodel.cpp

```diff
@@ -578,9 +578,9 @@ void WalletModel::getOutputs(const std::vector<COutPoint>& vOutpoints, std::vect
     for (const COutPoint& outpoint : vOutpoints)
     {
         if (!wallet->mapWallet.count(outpoint.hash)) continue;
-        int nDepth = wallet->mapWallet[outpoint.hash].GetDepthInMainChain();
+        int nDepth = wallet->mapWallet.at(outpoint.hash).GetDepthInMainChain();
         if (nDepth < 0) continue;
```

Comment:
Isn't this condition taking advantage of the dummy `CWalletTx`? If so replace `.at()` with `.find()` and test the iterator.